_posts/2021-05-22-fluxv2-openfaas.md

```diff
@@ -0,0 +1,332 @@
+
+We have the Flux cli we need to `bootstrap` the cluster and create a git repository. 
+
+For this section I'm using Github, you can find instructions for other Git providers [in the flux docs](https://fluxcd.io/docs/installation/#bootstrap)
+
+Generate a `github personal access token` with `repo` permissions [like this](https://docs.github.com/en/github/authenticating-to-github/creating-a-personal-access-token)
+
+Export this as an environment variable
+
+```sh 
+export GITHUB_TOKEN=<your-token>
+export GITHUB_USER=<your github username>
+```
+
+Run the bootstrap for a repository on your personal GitHub account:
+
+```sh 
+flux bootstrap github \
+  --owner=$GITHUB_USER \
+  --repository=openfaas-flux \
+  --path=clusters/linode \
+  --personal
+```
+
+After a short time we should see the flux controllers installed into the `flux-system` namespace!
+
+```sh 
+$ kubectl get pods -n flux-system
+NAME                                       READY   STATUS    RESTARTS   AGE
+helm-controller-85bfd4959d-w5ccr           1/1     Running   0          38s
+kustomize-controller-6977b8cdd4-gblq7      1/1     Running   0          38s
+notification-controller-5c4d48f476-7tdkk   1/1     Running   0          38s
+source-controller-85fb864746-gzrxq         1/1     Running   0          38s
+```
+
+This step has:
+* Installed the `flux controllers` into your cluster 
+* Created a github repository, named `openfaas-flux` (unless you changed that). 
+* Added a `Deploy key` to Github with access to that repository
+
+These flux controllers are using the deploy key to read changes in the github repository and install things based on 
+changes! In the next section we will add some `yaml` manifests into the git repository and see Flux install these into the cluster.
+
+# Install OpenFaaS
+
+Clone your git repository, for me the commands were
+
+```sh 
+git clone git@github.com:$GITHUB_USER/openfaas-flux.git
+cd openfaas-flux/clusters/linode
+```
+
+Our flux controllers are syncing manifests we change in the `clusters/linode` directory that we specified as the path on isntall.
+
+We can now add a `HelmRepository` and `HelmRelease` to install the OpenFaaS Chart
+
+You can get more information on the [flux components in the docs](https://fluxcd.io/docs/components/)
+
+Create our OpenFaaS namespaces by adding a file with these contents to the repository called `openfaas-namespaces.yaml`
+```yaml
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: openfaas
+  labels:
+    role: openfaas-system
+    access: openfaas-system
+---
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: openfaas-fn
+  labels:
+    role: openfaas-fn
+```
+
+Add a `HelmRepository` which tells Flux where it can find the OpenFaaS helm chart
+```sh 
+flux create source helm openfaas \ 
+  --url https://openfaas.github.io/faas-netes/ \
+  --export > openfaas-helmchart.yaml
```

Comment:
At this point what is the content of `openfaas-helmchart.yaml` and why wouldn't we print out the result of the command like we did everywhere else so far?

Reply:
It's empty - iv added a secton showing the contents